specification/azurefleet/AzureFleet.Management/fleet.tsp

```diff
@@ -211,6 +228,213 @@ union EvictionPolicy {
+  MemoryOptimized: "MemoryOptimized",
+
+  @doc("StorageOptimized VM.")
+  StorageOptimized: "StorageOptimized",
+
+  @doc("GpuAccelerated VM.")
+  GpuAccelerated: "GpuAccelerated",
+
+  @doc("FpgaAccelerated VM.")
+  FpgaAccelerated: "FpgaAccelerated",
+}
+
+@added(Versions.v2024_11_01)
+@doc("Architecture types supported by Azure VMs.")
+union CpuManufacturer {
+  string,
+
+  @doc("Intel CPU.")
+  Intel: "Intel",
+
+  @doc("AMD CPU.")
+  AMD: "AMD",
+}
+
+@added(Versions.v2024_11_01)
+@doc("Different kind of Local storage disk types supported by Azure VMs.")
+union LocalStorageDiskType {
+  string,
+
+  @doc("HDD DiskType.")
+  HDD: "HDD",
+
+  @doc("SDD DiskType.")
+  SSD: "SSD",
+}
+
+@added(Versions.v2024_11_01)
+@doc("Architecture types supported by Azure VMs.")
+union ArchitectureType {
+  string,
+
+  @doc("Intel Architecture")
+  Intel: "Intel",
+
+  @doc("AMD Architecture")
+  AMD: "AMD",
+}
+
+@added(Versions.v2024_11_01)
+@doc("Gpu manufacturers supported by Azure VMs.")
+union GpuManufacturer {
+  string,
+
+  @doc("Intel GpuType")
+  Intel: "Intel",
+
+  @doc("AMD GpuType")
+  AMD: "AMD",
+
+  @doc("Nvidia GpuType")
+  Nvidia: "Nvidia",
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "VMAttributes which does not follow camel casing"
+@added(Versions.v2024_11_01)
+@doc("VMAttributes that will be used to filter VMSizes which will be used to build Fleet.")
+model VMAttributes {
+  @doc("The range of vCpuCount specified from Min to Max. Must be specified if VMAttributes are specified, either Min or Max is required if specified.")
+  vCpuCount: VMAttributeMinMax;
+
+  @doc("The range of memory specified from Min to Max. Must be specified if VMAttributes are specified, either Min or Max is required if specified.")
+  memoryInMiB: VMAttributeMinMax;
+
+  @doc("Specifies whether the VMSize supporting local storage should be used to build Fleet or not.")
+  localStorageSupport?: VMAttributeSupport;
+
+  @extension("x-ms-identifiers", [])
+  @doc("The local storage disk types specified as a list. If gpuSupport is Excluded, this VMAttribute can not be used.")
+  localStorageDiskType: Array<LocalStorageDiskType>;
+
+  @doc("The range of local storage in GB specified from Min to Max.")
+  localStorageInGB: VMAttributeMinMax;
+
+  @extension("x-ms-identifiers", [])
+  @doc("The local storage disk types specified as a list.")
+  localStorageDiskTypes: Array<LocalStorageDiskType>;
+
+  @doc("The range of data disk count specified from Min to Max. Optional parameter. Either Min or Max is required if specified.")
+  dataDiskCount: VMAttributeMinMax;
+
+  @doc("The range of network interface count specified from Min to Max. Optional parameter. Either Min or Max is required if specified.")
+  networkInterfaceCount: VMAttributeMinMax;
+
+  @doc("The range of network bandwidth in Mbps specified from Min to Max. Optional parameter. Either Min or Max is required if specified.")
+  networkBandwidthInMbps: VMAttributeMinMax;
```

Comment:
All optional parameters should have `?`

Reply:
All properties are optional in `VMAttributes` except `vCpuCount: VMAttributeMinMax`, `memoryInMiB: VMAttributeMinMax`. Please make the change.

Reply:
Updated